### ISSUE TITLE
Consider inclusion of relativeToChangelogFile bugfix into 4.17.x release train

### ISSUE THREAD

**Issue by xak2000** [liquibase/liquibase#4147]
### Search first

- [X] I searched and no similar issues were found

### Description

In latest 4.17.x version (4.17.2 at the moment) there is bug that causes `relativeToChangelogFile="true"` to work incorrectly.

This bug is fixed in 4.18.0 (I'm not sure what commit fixes it).

The problems is: Spring Boot 3.0.x release train will not upgrade Liquibase to any version bigger than 4.17.x due to internal politics of not updating dependencies in patch releases of Spring Boot, except to updating to the version where only patch version of the same minor version of dependency is updated. So, Spring Boot mainteiners could not update to Liquibase 4.18+.

Also, Spring Boot 3.0.x is a major upgrade for many people with many changes, so migration from 2.x to 3.0.x is not an easy task by itself. And at the same time it happens to depend on 4.17.x version of Liquibase, that has this strange bug with relative paths.

This leads to the situation that after migration to Spring Boot 3.0.x and starting the application, many users will have very unexpected behavior when liquibase migrations will be not found anymore, or worse, be applied 2nd time with wrong migration paths saved to DB.

It would be good to have 4.17.3 version that will backprot fixes of these bugs, so could be applied to the next patch version of spring boot 3.0.x.

Related Sring Boot issue: spring-projects/spring-boot#35010.
More details about the bug in the [comment](https://github.com/liquibase/liquibase/issues/2281#issuecomment-1509080614).

### Steps To Reproduce

This is not a bug report, but a request to include already fixed bugs into previous minor version of Liquibase.

### Expected/Desired Behavior

The patch version 4.17.x is released that could be used by 3rd parties (e.g. Spring Boot) as bug-fix update.

### Liquibase Version

4.17.2

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by nvoxland** [comment]
Thanks for the background on how spring boot manages the dependency versions. I'll talk to the team about it

**Comment by nvoxland** [comment]
It looks like liquibase 4.20 will be released with spring boot 3.1.0 at the end of the month, correct?

https://github.com/spring-projects/spring-boot/blob/6a39b497ad898f431bedf7aa5fdc78e9e675e1a6/spring-boot-project/spring-boot-dependencies/build.gradle#L804

**Comment by xak2000** [comment]
That's right. I'm not sure about date, but yes, it will be Liquibase 4.20 (that includes the fix) in SB 3.1.0.

Though, the 3.0.x branch is still important as most users will migrate from 2.x.x to 3.0.x and only then to 3.1.x (or whatever version will be the latest at that moment). At least, I think this is the recommended migration path and most users will follow it: [Spring Boot 3.0 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide).

That is why I think it's important to include a stable Liquibase version with SB 3.0.x, if possible.

**Comment by filipelautert** [comment]
This backport request is no longer relevant — the fix was included in Liquibase 4.20, which shipped with Spring Boot 3.1.0. Spring Boot 3.0.x has since reached end of life. Thanks for the report!
